docs/agda-spec/src/Spec/Protocol.lagda

```diff
@@ -0,0 +1,172 @@
+
+\begin{itemize}
+  \item The stake pool stake distribution \afld{pd}.
+  \item The epoch nonce \afld{η₀}.
+\end{itemize}
+
+Its state is shown in Figure~\ref{fig:ts-types:prtcl} and consists of
+\begin{itemize}
+  \item The operational certificate issue number mapping \afld{cs}.
+  \item The evolving nonce \afld{ηv}.
+  \item The candidate nonce for the next epoch \afld{ηc}.
+\end{itemize}
+
+\begin{code}[hide]
+{-# OPTIONS --safe #-}
+
+open import Spec.BaseTypes using (Nonces)
+open import Spec.BlockDefinitions
+open import Ledger.Crypto
+open import Ledger.Script
+open import Ledger.Types.Epoch
+open import Data.Rational.Ext
+
+module Spec.Protocol
+  (crypto : _) (open Crypto crypto)
+  (nonces : Nonces crypto) (open Nonces nonces)
+  (es     : _) (open EpochStructure es)
+  (ss     : ScriptStructure crypto es) (open ScriptStructure ss)  
+  (bs     : BlockStructure crypto nonces es ss) (open BlockStructure bs)
+  (af     : _) (open AbstractFunctions af)
+  (rs     : _) (open RationalExtStructure rs)
+  where
+
+open import InterfaceLibrary.Common.BaseTypes crypto using (PoolDistr; lookupPoolDistr)
+open import Spec.OperationalCertificate crypto nonces es ss bs af
+open import Spec.UpdateNonce crypto nonces es
+open import Spec.BaseTypes crypto using (OCertCounters)
+open import Data.Rational as ℚ using (ℚ; 0ℚ; 1ℚ)
+open import Ledger.Prelude
+open Ledger.Prelude.ℤ using (pos)
+
+\end{code}
+
+\begin{figure*}[h]
+\emph{Protocol environments}
+\begin{code}
+record PrtclEnv : Type where
+  constructor ⟦_,_⟧ᵖᵉ
+  field pd  : PoolDistr -- pool stake distribution
+        η₀  : Nonce     -- epoch nonce
+\end{code}
+\emph{Protocol states}
+\begin{code}
+record PrtclState : Type where
+  constructor ⟦_,_,_⟧ᵖˢ
+  field cs : OCertCounters -- operational certificate issues numbers
+        ηv : Nonce         -- evolving nonce
+        ηc : Nonce         -- candidate nonce
+\end{code}
+\emph{Protocol transitions}
+\begin{code}[hide]
+data
+\end{code}
+\begin{code}
+  _⊢_⇀⦇_,PRTCL⦈_ : PrtclEnv → PrtclState → BHeader → PrtclState → Type
+\end{code}
+\caption{Protocol transition system types}
+\label{fig:ts-types:prtcl}
+\end{figure*}
+
+\begin{figure*}[h]
+\emph{Protocol helper functions}
+\begin{code}
+hBLeader : BHBody → ∃[ n ] n < 2 ^ 512 -- [0, 2^512)
+hBLeader bhb = serHashToℕ (hash (encode "L" ∥ encode vrfRes))
+  where open BHBody bhb
+
+hBNonce : BHBody → Nonce
+hBNonce bhb = serHashToNonce (hash (encode "N" ∥ encode vrfRes))
+  where open BHBody bhb
+
+checkLeaderVal : ∃[ n ] n < 2 ^ 512 → PosUnitInterval → ℚ → Type
+checkLeaderVal (certℕ , certℕprf) (f , posf , f≤1) σ =
+   if f ≡ 1ℚ then ⊤ else λ{f≢1ℚ} →
+     let
+       p     = pos certℕ ℚ./ (2 ^ 512)
+       p≢1ℚ  = ↥p<↧p⇒p≢1 {p} (n<m⇒↥[n/m]<↧[n/m] certℕprf)
+       q     = 1ℚ ℚ.- p
+       1-f≥0 = p≤1⇒1-p≥0 f≤1
+       1-f≢0 = p≢1⇒1-p≢0 f≢1ℚ
+       instance
+         q≢0ℚ  = ℚ.≢-nonZero (p≢1⇒1-p≢0 p≢1ℚ)
+         1-f>0 = ℚ.positive (≤∧≢⇒< 1-f≥0 $ ≢-sym 1-f≢0)

```

Comment:
As already discussed, I think you should hide this. You could also put `λ{f≢1ℚ} →` on a separate line and hide it as well then.

Reply:
Fixed in 893f3d9579b75ea43259557f8eda2a9d13e27834.